app/shared/environment.js

```diff
@@ -0,0 +1,49 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+/* global LIBDIR */
+
+import yargs from 'yargs';
+import path from 'path';
+import AppDirectory from 'appdirectory';
+import fs from 'fs-extra';
+import manifest from '../../package.json';
+import BUILD_CONFIG from '../../build-config.json';
+
+export const argparser = yargs.usage('Usage: $0 [options]')
```

Comment:
I believe usages of this module are currently split across process boundaries, but having them modify the state of an exported variable from a shared module seems like a possibility for a bug in the future (i.e. the next time this module is required it's going to have the options from the previous one included).

I'm not sure the best API to avoid that here, but maybe something like:

    export function getArgParser()
      return yargs.usage(...).option(...);
    }

    export function parseArgs(argparser) {
      const options = argparser.argv;
      ...
    }